caching_options.hh

```diff
@@ -59,8 +62,17 @@ class caching_options {
     caching_options() : _key_cache(default_key), _row_cache(default_row) {}
 public:
 
+    bool enabled() const {
+        return _enabled;
+    }
+
     std::map<sstring, sstring> to_map() const {
-        return {{ "keys", _key_cache }, { "rows_per_partition", _row_cache }};
+        std::map<sstring, sstring> res = {{ "keys", _key_cache },
+                { "rows_per_partition", _row_cache }};
+        if (!_enabled) {
+            res.insert({"enabled", "false"});
```

Comment:
Why is this conditional?

Reply:
Because missing parameter means enabled == true. That's just to keep it more backward compatible.

Reply:
I think we should add it unconditionally, otherwise all code wishing to use this map has to learn about what a missing `enabled` means.

Reply:
The code will have to do it anyway because the parameter is missing on older nodes and then it means enabled == true.

Reply:
Still, why complicate this method for no gain? We don't need this to be backward compatible as older nodes will not see this option anyway, will they?

Reply:
Well they shouldn't see it because of the feature but I think it's safer for rollback or stuff like this. This is not big complication and actually gives us more safety. For example if no table has disabled cache, it is like the feature does not exist at all. zero-overhead feature ;)

Reply:
Ok.

Reply:
Agree it's safer. It also allows a DESCRIBE TABLE on a new cluster to be replayed on an old cluster with fewer surprises.

Reply:
It's also better to preserve the schema digest.

Reply:
But, something is missing here. If the user specifed enabled: true explicitly, then when we call to_map() we get it back without the option.

You can fix by making _enabled an optional<boo> (and return value_or(true) from enabled()).

Reply:
And this can lead to schema digest mismatch. Please add a test that fails on the current patch and succeeds on the fixed patch.

Reply:
Could you explain more how this can create a schema digest mismatch @avikivity? I don't see it. No matter whether enabled was set explicitly or wasn't set at all, exchanged schema will have no 'enabled' element in the map. Or am I missing something?

Reply:
> But, something is missing here. If the user specifed enabled: true explicitly, then when we call to_map() we get it back without the option.
> 
> You can fix by making _enabled an optional (and return value_or(true) from enabled()).

Setting 'enabled' to true and not setting means the same. Is it really a problem?

Reply:
I think you are correct. I thought the map could get into schema_tables directly, but it always goes through to_map(), so enabled: true will always be filtered out.